script/CommonDeployer.s.sol

```diff
@@ -37,18 +37,28 @@ struct CommonReport {
 }
 
 contract CommonActionBatcher {
-    // @dev lock to ensure we forbid calling engage/revoke methods after deployment
-    bool isLock;
+    error NotDeployer();
 
-    modifier unlocked() {
-        if (!isLock) _;
+    address deployer;
+
+    constructor() {
+        deployer = msg.sender;
+    }
+
+    modifier onlyDeployer() {
+        require(msg.sender == deployer, NotDeployer());
+        _;
+    }
+
+    function setDeployer(address newDeployer) public onlyDeployer {
+        deployer = newDeployer;
     }
```

Comment:
This is needed for testing, when the deployer of the ActionBatcher can not be the sender